queries.sql

```diff
@@ -86,4 +86,66 @@ WHERE escape_attempts = 0 AND full_name = 'Dean Winchester';
\ No newline at end of file
+GROUP BY full_name ORDER BY COUNT(*) DESC LIMIT 1;
+
+
+-- Queries with Link Tables
+
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.name AS animal_name, vets.name AS vet_name, date_visit FROM visits
+JOIN vets ON visits.vets_id = vets.id
+JOIN animals ON visits.animals_id = animals.id
+WHERE vets.name = 'William Tatcher' ORDER BY date_visit DESC LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT animals.name AS animal_name, vets.name AS vet_name FROM visits
+JOIN vets ON vets.id = visits.vets_id
+JOIN animals ON animals.id = visits.animals_id
+WHERE vets.name = 'Stephanie Mendez';
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.name AS vet_name , species.name AS species_name FROM vets
+LEFT JOIN specializations ON specializations.vets_id = vets.id
+LEFT JOIN species ON species.id = specializations.species_id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name AS animal_name, vets.name AS vet_name, date_visit FROM animals
+JOIN visits ON animals.id = visits.animals_id
+JOIN vets ON vets.id =  visits.vets_id
+WHERE vets.name = 'Stephanie Mendez' AND date_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+-- What animal has the most visits to vets?
+SELECT animals.name AS animal_name, COUNT(visits.animals_id) FROM animals
+JOIN visits ON animals.id = visits.animals_id
+GROUP BY animals.name ORDER BY COUNT(visits.animals_id) DESC LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+SELECT vets.name AS vet_name, animals.name AS animal_name, date_visit FROM animals
+JOIN visits ON visits.animals_id = animals.id
+JOIN vets ON vets.id = visits.vets_id
+WHERE vets.name = 'Maisy Smith' ORDER BY date_visit ASC LIMIT 1;
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT animals.name AS animal_name, animals.date_of_birth AS animal_DOB,escape_attempts,weight_kg AS animal_weight, vets.name AS vet_name, vets.age AS vet_age, date_visit FROM vets
+FULL OUTER JOIN visits ON visits.animals_id = vets.id
+JOIN animals ON animals.id = visits.vets_id
+ORDER BY date_visit DESC LIMIT 1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT COUNT(visits.vets_id), vets.name AS vet_name FROM vets
+FULL OUTER JOIN visits ON visits.vets_id = vets.id
+FULL OUTER JOIN specializations ON specializations.vets_id = vets.id
+WHERE specializations.species_id IS NULL
+GROUP BY vets.name;
+
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+
+SELECT species.name , COUNT(visits.date_visit) AS visit_count
+FROM animals
+JOIN species ON species.id = animals.species_id
+JOIN visits ON visits.animals_id = animals.id
+JOIN vets ON vets.id = visits.vets_id
+WHERE vets.name = 'Maisy Smith'
+GROUP BY species.name
+ORDER BY COUNT(species_id) DESC LIMIT 1;
```

Comment:
- [ ] Some of your queries have wrong results. and the last reviewer has not mentioned it because he asked you to add the data in the visits table. now based on this change, I think you should have updated your screenshots with newer results.
Now here in this query, your result is empty but in reality Maisy Smiths should consider getting speciality in `Digimon as she has 6 visits by Digimons` so consider fixing it.
your result
![t](https://github.com/Zuhaib042/vet-clinic-database/assets/96683843/116df013-2050-4a89-9b87-2c104da6972f)